setup/Microsoft.VisualStudio.ProjectSystem.Managed.CommonFiles/Microsoft.VisualStudio.ProjectSystem.Managed.CommonFiles.csproj

```diff
@@ -1,24 +1,39 @@
 ﻿<!-- Licensed to the .NET Foundation under one or more agreements. The .NET Foundation licenses this file to you under the MIT license. See the LICENSE.md file in the project root for more information. -->
 <Project Sdk="Microsoft.NET.Sdk">
+
   <Import Project="..\..\build\import\VisualStudio.props" />
-  
+
   <PropertyGroup>
     <!-- VS Insertion -->
     <VisualStudioInsertionComponent>Microsoft.VisualStudio.ProjectSystem.Managed</VisualStudioInsertionComponent>
   </PropertyGroup>
+
   <ItemGroup>
     <!-- Depend on projects producing XAML rules included in this Willow package -->
     <ProjectReference Include="..\..\src\Microsoft.VisualStudio.ProjectSystem.Managed\Microsoft.VisualStudio.ProjectSystem.Managed.csproj" />
   </ItemGroup>
+
   <ItemGroup>
     <SwrProperty Include="Version=$(VsixVersion)" />
     <SwrProperty Include="VisualStudioXamlRulesDir=$(VisualStudioXamlRulesDir)" />
   </ItemGroup>
+
   <ItemGroup>
     <SwrFile Include="*.swr" />
   </ItemGroup>
 
   <Target Name="Build" DependsOnTargets="ResolveProjectReferences">
     <MakeDir Directories="$(IntermediateOutputPath)" ContinueOnError="True" />
   </Target>
+
+  <!-- Adds the appropriate vsix files for signing. -->
+  <Target Name="AddVsixForSigning" DependsOnTargets="SwixBuild" BeforeTargets="SignFiles">
+    <Copy SourceFiles="$(VisualStudioSetupInsertionPath)$(AssemblyName).vsix" DestinationFolder="$(OutDir)" />
+    <ItemGroup>
+      <FilesToSign Include="$(OutDir)$(AssemblyName).vsix">
+        <Authenticode>VsixSHA2</Authenticode>
+      </FilesToSign>
+    </ItemGroup>
+  </Target>
```

Comment:
It will look like that this target is copied 3 times among the setup projects. However, this particular target has a different `DependsOnTargets` than the other two. The way this Vsix is generated is different than the other two as it relies on a process from RepoToolset. The other projects don't seem to follow this process. I didn't dig too deep here but that is what the build logs/binlogs show. This will need to get investigated again when transitioning off of RepoToolset.

Reply:
One common pattern that avoids this kind of duplication is to specify the `DependsOnTargets` as a property, such as `$(AddVsixForSigningDependsOnTargets)`. Perhaps you can use that here too.

Reply:
I ended up doing this and set that property in the corresponding projects. Now, this target is in the `setup/Directory.Build.targets`.